From 36a070be1f8ec6ad3a2e80c3cba16407a6d9ff35 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 12 Oct 2022 13:07:43 +0300
Subject: [PATCH] Revert "[SW-106502] gaudi3: perform mmu bypass configuration"

This reverts commit 73b0d1b5799c123817f545e6229ad244a4f66317.

mmu bypass configuration default values fix is added in RTL
so ROM configuration is not required.

Change-Id: If79744ef647c70246f5a7b98e1264a8724cb32b2
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 0b38d86..86f82b7 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -453,13 +453,11 @@
 }
 
 __weak void arcboot_rom_init_boot_info(void) {}
-__weak void arcboot_rom_set_mmu_bp(void) {}
 
 int arcboot_rom_main(void)
 {
 	arcboot_rom_init();
 	arcboot_rom_fuse();
-	arcboot_rom_set_mmu_bp();
 	arcboot_rom_security_init();
 	arcboot_rom_disable_spi_gw();
 	arcboot_rom_wait_for_signal();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 5cedf2f..093aa4d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -478,30 +478,6 @@
 		arcboot_rom_persist_data_non_zero_init();
 }
 
-void arcboot_rom_set_mmu_bp(void)
-{
-	uint32_t pcie_ctrl_val = arcboot_rom_fuse_pcie_ctrl_config();
-
-	regmap_writel((D0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_WR_M |
-		D0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_RD_M),
-		mmD0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS);
-
-	regmap_writel((D0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD_WR_M |
-		D0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD_RD_M),
-		mmD0_PARC_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD);
-
-	if (arcboot_rom_global.die_id && !pcie_ctrl_val)
-		return;
-
-	regmap_writel((D0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_WR_M |
-		D0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_RD_M),
-		mmD0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS);
-
-	regmap_writel((D0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD_WR_M |
-		D0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD_RD_M),
-		mmD0_PCIE_MSTR_RR_MSTR_IF_AXUSER_HBW_HB_MMU_BYPASS_OVRD);
-}
-
 void arcboot_rom_init(void)
 {
 	arcboot_rom_persistent_dccm_init();
